docs/DevGuide/Parallelization.md

```diff
@@ -0,0 +1,539 @@
+\cond NEVER
+Distributed under the MIT License.
+See LICENSE.txt for details.
+\endcond
+# Parallization, Charm++, and Core Concepts {#dev_guide_parallelization_foundations}
+
+\tableofcontents
+
+# Introduction {#dev_guide_parallelization_introduction}
+
+SpECTRE builds a layer on top of Charm++ that performs various safety checks and
+initialization for the user that can otherwise lead to difficult-to-debug
+undefined behavior. The central concept is what is called a %Parallel
+Component. A %Parallel Component is a struct with several type aliases that
+is used by SpECTRE to set up the Charm++ chares and allowed communication
+patterns. %Parallel Components are input arguments to the compiler, which then
+writes the parallelization infrastructure that you requested for the executable.
+There is no restriction on the number of %Parallel Components, though
+practically it is best to have around 10 at most.
+
+Here is an overview of what is described in detail in the sections below:
+
+- \ref dev_guide_parallelization_metavariables_class "Metavariables": Provides
+  high-level configuration to the compiler, e.g. the physical system to be
+  simulated.
+- \ref dev_guide_parallelization_phases_of_execution "Phases": Defines distinct
+  simulation phases separated by a global synchronization point,
+  e.g. `%Initialization`, `Evolve` and `Exit`.
+- \ref dev_guide_parallelization_core_algorithm "Algorithm": In each phase,
+  iterates over a list of actions until the current phase ends.
+- \ref dev_guide_parallelization_parallel_components "Parallel component":
+  Maintains and executes its algorithm.
+- \ref dev_guide_parallelization_actions "Action": Performs a computational
+  task, e.g. evaluating the right hand side of the time evolution equations. May
+  require data to be received from another action potentially being executed on
+  a different core or node.
+
+# The Metavariables Class {#dev_guide_parallelization_metavariables_class}
+
+SpECTRE takes a different approach to input options passed to an executable than
+is common. SpECTRE not only reads an input file at runtime but also has many
```

Comment:
I actually think it's important to emphasize the difference since all scientific codes I've used either hard code everything or read everything from an input file.